build/Dockerfile

```diff
@@ -1,46 +1,289 @@
+	&& echo "Acquire::https::plus-pkgs.nginx.com::User-Agent  \"k8s-ic-$IC_VERSION-apt\";" >> /etc/apt/apt.conf.d/90nginx \
+	&& printf "deb https://plus-pkgs.nginx.com/debian buster nginx-plus\n" > /etc/apt/sources.list.d/nginx-plus.list \
+	&& apt-get purge --auto-remove -y gnupg wget \
+	&& rm -rf /var/lib/apt/lists/*
+
+RUN --mount=type=secret,id=nginx-repo.crt,dst=/etc/ssl/nginx/nginx-repo.crt,mode=0644 \
+	--mount=type=secret,id=nginx-repo.key,dst=/etc/ssl/nginx/nginx-repo.key,mode=0644 \
+	apt-get update && apt-get install -y apt-transport-https libcap2-bin nginx-plus=${NGINX_PLUS_VERSION} nginx-plus-module-njs=${NGINX_NJS_VERSION} \
+	&& apt-get purge --auto-remove -y apt-transport-https \
+	&& rm -rf /var/lib/apt/lists/*
+
+COPY --chown=nginx:0 internal/configs/oidc/* /etc/nginx/oidc/
+
+
+############################################# Base image for Debian with NGINX Plus and App Protect #############################################
+FROM debian-plus as debian-plus-ap-intermediate
+ARG IC_VERSION
+LABEL version=debian-nginx-plus-ap-$IC_VERSION
+LABEL maintainer="NGINX Docker Maintainers <docker-maint@nginx.com>"
+
+ENV APPPROTECT_MODULE_VERSION 23+3.281.0-1~buster
+ENV APPPROTECT_PLUGIN_VERSION 3.281.0-1~buster
+ENV APPPROTECT_ENGINE_VERSION 5.18.0-1~buster
+ENV APPPROTECT_COMPILER_VERSION 5.18.0-1~buster
+ENV APPPROTECT_SIG_VERSION 2020.12.30-1~buster
+ENV APPPROTECT_THREAT_CAMPAIGNS_VERSION 2021.01.03-1~buster
+
+RUN --mount=type=secret,id=nginx-repo.crt,dst=/etc/ssl/nginx/nginx-repo.crt,mode=0644 \
+	--mount=type=secret,id=nginx-repo.key,dst=/etc/ssl/nginx/nginx-repo.key,mode=0644 \
+	set -x \
+	&& apt-get update \
+	&& apt-get install --no-install-recommends --no-install-suggests -y gnupg wget \
+	&& wget https://nginx.org/keys/app-protect-sigs.key \
+	&& gpg --no-default-keyring --keyring app_protect_keyring.gpg --import app-protect-sigs.key \
+	&& gpg --no-default-keyring --keyring app_protect_keyring.gpg --export > /etc/apt/trusted.gpg.d/nginx_app_signing.gpg \
+	&& sed -i '$ d' /etc/apt/apt.conf.d/90nginx \
+	&& echo "Acquire::https::plus-pkgs.nginx.com::User-Agent  \"k8s-ic-$IC_VERSION-app-$APPPROTECT_VERSION-apt\";" >> /etc/apt/apt.conf.d/90nginx \
+	&& echo "deb https://app-protect-sigs.nginx.com/debian/ buster nginx-plus\n" | tee /etc/apt/sources.list.d/app-protect-sigs.list \
+	&& echo "Acquire::https::app-protect-sigs.nginx.com::Verify-Peer \"true\";" >> /etc/apt/apt.conf.d/90app-protect-sigs \
+	&& echo "Acquire::https::app-protect-sigs.nginx.com::Verify-Host \"true\";" >> /etc/apt/apt.conf.d/90app-protect-sigs \
+	&& echo "Acquire::https::app-protect-sigs.nginx.com::SslCert     \"/etc/ssl/nginx/nginx-repo.crt\";" >> /etc/apt/apt.conf.d/90app-protect-sigs \
+	&& echo "Acquire::https::app-protect-sigs.nginx.com::SslKey      \"/etc/ssl/nginx/nginx-repo.key\";" >> /etc/apt/apt.conf.d/90app-protect-sigs \
+	&& apt-get purge --auto-remove -y gnupg wget \
+	&& rm -rf /var/lib/apt/lists/*
```

Comment:
what about files in /etc/apt/apt.conf.d like https://github.com/nginxinc/kubernetes-ingress/blob/master/build/appprotect/DockerfileWithAppProtectForPlus ?